app/models/filter.rb

```diff
@@ -201,7 +201,14 @@ def parenthesize(string)
   # if we have 0 types, empty validation will set error, we can't have more than one type
   def same_resource_type_permissions
     types = self.permissions.map(&:resource_type).uniq
-    errors.add(:permissions, _('must be of same resource type (%s)') % types.join(',')) if types.size > 1
+    errors.add(
```

Comment:
This is not strictly needed but it makes debugging easier - if a filter fails to save, I want to know which Role is it associated with (it's easier to reason about, I can tell where a role is coming, like "Discovery Reader", but filters have no name)